scripts/enumerable.rb

```diff
@@ -0,0 +1,209 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return return_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return return_enum unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    select = []
+    my_each { |item| select << item if yield item }
+    select
+  end
+
+  def my_all?(class_passed = nil)
+    all = true
+    if block_given?
+      my_each { |item| all = false unless yield item }
+    elsif class_passed
+      my_each { |item| all = false unless item.is_a? class_passed }
+    else
+      my_each { |item| all = false unless item }
+    end
+    all
+  end
```

Comment:
`my_all?` should not only accept a `Class` but generally a `pattern` or a `block`, By `pattern`, it could be a `Class`, a `RegExp`, `String`, `Integer` and so on.

This is the same for `my_any?` and `my_none?` 

Find more details about [`all?` here](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F)

   ```ruby
   # When an Integer is passed
   irb(main):006:0> [2,2,2].all?(2)
   => true
   irb(main):007:0> [2,2,2].my_all?(2)
   TypeError (class or module required)

   # When a RegEx pattern is passed
   irb(main):018:0> %w[ant bat cat].all?(/t/)
   => true
   irb(main):019:0> %w[ant bat cat].my_all?(/t/)
   TypeError (class or module required)
   ```
   #### :bulb:  Hint: Use ` === ` to check if a member matches a patern.